src/bot/master.rs

```diff
@@ -192,22 +173,32 @@ impl MasterBot {
     }
 
     async fn spawn_bot_for(&self, id: ClientId) {
-        if let Some(bot_args) = self.build_bot_args_for(id) {
-            let (bot, fut) = MusicBot::new(bot_args).await;
-            tokio::spawn(fut.unit_error().boxed().compat().map(|_| ()));
-            let mut music_bots = self.music_bots.write().expect("RwLock was not poisoned");
-            music_bots
-                .connected_bots
-                .insert(bot.name().to_string(), bot);
+        match self.build_bot_args_for(id) {
+            Ok(bot_args) => {
+                let (bot, fut) = MusicBot::new(bot_args).await;
+                tokio::spawn(fut.unit_error().boxed().compat().map(|_| ()));
+                let mut music_bots = self.music_bots.write().expect("RwLock was not poisoned");
+                music_bots
+                    .connected_bots
+                    .insert(bot.name().to_string(), bot);
+            }
+            Err(e) => self.teamspeak.send_message_to_user(id, &e.to_string()),
         }
     }
 
     async fn on_message(&self, message: MusicBotMessage) -> Result<(), AudioPlayerError> {
-        if let MusicBotMessage::TextMessage(message) = message {
-            if let MessageTarget::Poke(who) = message.target {
-                info!("Poked by {}, creating bot for their channel", who);
-                self.spawn_bot_for(who).await;
+        match message {
+            MusicBotMessage::TextMessage(message) => {
+                if let MessageTarget::Poke(who) = message.target {
+                    info!("Poked by {}, creating bot for their channel", who);
+                    self.spawn_bot_for(who).await;
+                }
+            }
+            MusicBotMessage::ChannelCreated(_) => {
+                // TODO Only subscribe to one channel
+                self.teamspeak.subscribe_all();
```

Comment:
This is a bit ugly but to subscribe to only one channel a new method has to be added to `tsclientlib` first.